runtime/macros/src/entry.rs

```diff
@@ -51,3 +51,26 @@ pub fn main(args: TokenStream, input: TokenStream) -> Result<TokenStream, Error>
         }
     })
 }
+
+pub fn profile(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    let input: ItemFn = syn::parse2(item).unwrap();
+    let visibility = &input.vis;
+    let name = &input.sig.ident;
+    let inputs = &input.sig.inputs;
+    let output = &input.sig.output;
+    let block = &input.block;
+    let generics = &input.sig.generics;
+    let where_clause = &input.sig.generics.where_clause;
+
+    let result = quote! {
+        #visibility fn #name #generics (#inputs) #output #where_clause {
+            profile_cycles!("^:{}", stringify!(#name));
+            let result = (|| #block)();
+            profile_cycles!("$:{}", stringify!(#name));
+            result
+
+        }
+    };
+
+    result
```

Comment:
1. this module is called `entry`, which is for the entry function. To keep it clean, let's move it to a separate one.
2. Just to be more strict, let's return an error if `attr` is not empty, like in `main`. Alternatively, users should be able to specify custom name for the function to avoid name clash. Or this macro should specify its full path.
3. use absolute path to external methods, like `nexus_rt::profile_cycles`. `nexus_rt` (very unlikely) but can happen to be renamed. https://crates.io/crates/proc-macro-crate
4. You should only unwrap in procedural macros when you can prove that it's practically unreachable. Making this macro to panic is very easy, a compile error should be returned instead.

Finally, this can be simplified.
```rust
let nexus_rt = {
    // get the import name
};
let input: ItemFn = syn::parse2(item)?;
let vis = &input.vis;
let block = &input.block;
let fn_name = &input.sig.ident.to_string();

quote! {
    #vis #sig {
        #nexus_rt :: profile_cycles!("^:{}", fn_name);
        let result = #block;
        #nexus_rt :: profile_cycles!("$:{}", fn_name);
        result
    }
}.into()
```

Reply:
Simplified and use `proc-macro-crate` as you suggested in [6dd69d0](https://github.com/nexus-xyz/nexus-zkvm/pull/243/commits/6dd69d0f13e127f638cd2f061f65cf167686c5e9)